src/components/Counter/Counter.jsx

```diff
@@ -1,9 +1,26 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState, useReducer } from 'react'
 
 const pinkRGB = `rgb(236, 72, 153)`
+const startValue = 0
+
+function countReducer(count, action) {
+  switch (action.type) {
+    case 'increment': {
+      return count + 1
+    }
+    case 'decrement': {
+      return count - 1
+    }
+    case 'reset': {
+      return 0
+    }
+    default:
+      throw new Error()
+  }
+}
 
 export default function Counter() {
-  const [count, setCount] = useState(0)
+  const [count, dispatch] = useReducer(countReducer, startValue)
   const [currentColor, setCurrentColor] = useState(pinkRGB)
```

Comment:
You can also add the color to your reducer too to handle both states in one place